### PR TITLE
feature/eugene/dat less/update regression tests with qa etl

### PR DESCRIPTION
**Pull request by eugsokolov**

### CODE REVIEW

File: datamonster_api/lib/company.py [hunk 1]
```diff
@@ -26,6 +26,9 @@ def __repr__(self):
         ticker_str = "[{}] ".format(self.ticker) if self.ticker else ""
         return "<{}: {}{}>".format(self.__class__.__name__, ticker_str, self.name)
 
+    def __eq__(self, obj):
+        return isinstance(obj, Company) and self.get_details() == obj.get_details()
```
Comment (jlindenger) [inline datamonster_api/lib/company.py]: Does it make sense to check equality against the ids (section_pks) instead?

File: datamonster_api/lib/datasource.py [hunk 2]
```diff
@@ -20,6 +20,9 @@ def __init__(self, _id, name, category, uri, dm):
         self.uri = uri
         self.dm = dm
 
+    def __eq__(self, obj):
+        return isinstance(obj, Datasource) and self.get_details() == obj.get_details()
```
Comment (jlindenger) [inline datamonster_api/lib/datasource.py]: Does it make sense to check equality against the ids (data fountain uuids) instead?

File: datamonster_api/lib/company.py (hunk 1 above)
Comment (eugsokolov) [reply datamonster_api/lib/company.py]: updated

File: datamonster_api/lib/datasource.py (hunk 2 above)
Comment (eugsokolov) [reply datamonster_api/lib/datasource.py]: updated

File: datamonster_api/lib/company.py [hunk 3]
```diff
@@ -27,7 +27,7 @@ def __repr__(self):
         return "<{}: {}{}>".format(self.__class__.__name__, ticker_str, self.name)
 
     def __eq__(self, obj):
```
Comment (jlindenger) [inline datamonster_api/lib/company.py]: Would it make sense to move this to BaseClass and have it be something like:
```
def __eq__(self, obj):
        return isinstance(obj, self.__class__) and self.id == obj.id
```

**Comment by eugsokolov** [comment]
the QA etl does not have static historical data, we are adding a new df to be static

**Comment by eugsokolov** [comment]
```
datamonster_api/tests/lib/regression.py .........                                                                                                          [100%]

=================================================================== 9 passed in 35.59 seconds ====================================================================
```

File: datamonster_api/tests/lib/test_company.py [hunk 4]
```diff
@@ -10,6 +10,11 @@ def _assert_object_matches_company(company, company_obj):
     assert company_obj["uri"] == company.uri
 
 
+def test_equality(company, other_company):
+    assert company == company
```
Comment (ted-brundage-am) [inline datamonster_api/tests/lib/test_company.py]: When would this not be true?

File: datamonster_api/tests/lib/test_datamonster.py [hunk 5]
```diff
@@ -13,6 +12,11 @@ def _assert_object_matches_datasource(datasource, datasource_obj):
     assert datasource_obj["uri"] == datasource.uri
 
 
+def test_equality(datasource, other_datasource):
+    assert datasource == datasource
```
Comment (ted-brundage-am) [inline datamonster_api/tests/lib/test_datamonster.py]: Same -- why do we check this assert?

File: datamonster_api/tests/lib/regression.py [hunk 6]
```diff
@@ -143,24 +223,22 @@ def test_data_source_2():
...
-def test_bigger_data_source():
-    """ Test SimilarWeb which is a bigger dataset
+def test_get_data_bigger():
+    """ Test `SimilarWeb Direct Volume` which is a bigger dataset
     """
     company = dm.get_company_by_id(335)
-    agg = Aggregation(period="week", company=company)
-
     ds = dm.get_datasource_by_id("5899e237-874c-4e77-9d2e-c4b6cff218e8")
-    assert ds.name == "SimilarWeb Direct Volume"
```
Comment (ted-brundage-am) [inline datamonster_api/tests/lib/regression.py]: Why aren't we checking the name?

File: datamonster_api/tests/lib/regression.py [hunk 7]
```diff
@@ -143,24 +223,22 @@ def test_data_source_2():
     assert_frame_equal(df.head(1), pandas.DataFrame.from_records([records]))
 
 
-def test_bigger_data_source():
-    """ Test SimilarWeb which is a bigger dataset
+def test_get_data_bigger():
+    """ Test `SimilarWeb Direct Volume` which is a bigger dataset
```
Comment (ted-brundage-am) [inline datamonster_api/tests/lib/regression.py]: Is this not static data? If it is, seems like we should add some stronger assertions on the df contents.

File: datamonster_api/tests/lib/regression.py [hunk 8]
```diff
@@ -109,15 +191,13 @@ def test_data_source():
...
 
-def test_data_source_2():
-    """ Test FactSet Actuals Sales Quarterly, one of the more popular data sets
+def test_get_data_factset():
+    """ Test `FactSet Actuals Sales Quarterly` one of the more popular data sets
     """
     company = dm.get_company_by_id(335)
-
     sales = next(dm.get_datasources(query="FactSet Actuals Sales Quarterly"))
     assert sales.id == "bdcac6ae-4f31-4aaf-a92a-12854f09c768"
```
Comment (ted-brundage-am) [inline datamonster_api/tests/lib/regression.py]: Should these UUIDs be saved off as constants somewhere?

File: datamonster_api/tests/lib/test_company.py (hunk 4 above)
Comment (eugsokolov) [reply datamonster_api/tests/lib/test_company.py]: if someone makes a code change to break the equality logic (or the conftest itself)

File: datamonster_api/tests/lib/regression.py (hunk 6 above)
Comment (eugsokolov) [reply datamonster_api/tests/lib/regression.py]: the point here was just to check that we can pull in a larger data set - the name property is checked on the QA ETL (and is repetitive/annoying to check all the time)

File: datamonster_api/tests/lib/regression.py (hunk 7 above)
Comment (eugsokolov) [reply datamonster_api/tests/lib/regression.py]: the point here was just to check that we can pull in a larger data set - the stronger assertions are done on the QA ETL and factset

File: datamonster_api/tests/lib/regression.py (hunk 8 above)
Comment (eugsokolov) [reply datamonster_api/tests/lib/regression.py]: ok

File: datamonster_api/tests/lib/test_company.py (hunk 4 above)
Comment (ted-brundage-am) [reply datamonster_api/tests/lib/test_company.py]: sure

File: datamonster_api/tests/lib/regression.py (hunk 6 above)
Comment (ted-brundage-am) [reply datamonster_api/tests/lib/regression.py]: kk

File: datamonster_api/tests/lib/regression.py (hunk 7 above)
Comment (ted-brundage-am) [reply datamonster_api/tests/lib/regression.py]: kk

**Pull request merged**